homeassistant/components/xiaomi_miio/gateway.py

```diff
@@ -170,3 +178,25 @@ def available(self):
             return False
 
         return self.coordinator.data[ATTR_AVAILABLE]
+
+    @callback
+    def push_callback(self, action, params):
+        """Push from subdevice."""
+        _LOGGER.debug("Got new push_callback: %s, %s", action, params)
+        self.schedule_update_ha_state()
+        self.hass.bus.fire(
```

Comment:
Entities should not fire custom events.

Reply:
This is removed and replaced by device triggers.